### PR TITLE
Add check for wrong keyword in starred dict

### LINKED ISSUE #782: Forbid to unpack values like `**{'@': 1}`
# Rule request

## Thesis

Related to #767

This should be forbidden: 

```
>>> print(**{'@': 2})
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: '@' is an invalid keyword argument for print()
```

To validate this we need to use `str.is_identifier`

## Reasoning

This is simply wrong.

### PR DESCRIPTION
**Pull request by ea-volkov**
# I have made things!

## Checklist

- [x] I have double checked that there are no unrelated changes in this pull request (old patches, accidental config files, etc)
- [x] I have created at least one test case for the changes I have made
- [ ] I have updated the documentation for the changes I have made
- [ ] I have added my changes to the `CHANGELOG.md`

## Related issues

Closes #782

### CODE REVIEW

File: tests/test_checker/test_noqa.py [hunk 1]
```diff
@@ -189,6 +189,7 @@
     'WPS519': 1,
     'WPS520': 1,
     'WPS521': 1,
+    'WPS522': 0,
```
Comment (sobolevn) [inline tests/test_checker/test_noqa.py]: Please, include at least one error in `noqa.py`

File: tests/test_visitors/test_ast/test_statements/test_wrong_named_keyword.py [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    WrongNamedKeywordViolation,
+)
+from wemake_python_styleguide.visitors.ast.statements import (
+    WrongNamedKeywordVisitor,
+)
+
+
+@pytest.mark.parametrize('code', [
+    'print(**{"@": "2"})',
+    'print(**{"2ab": "2"})',
```
Comment (sobolevn) [inline tests/test_visitors/test_ast/test_statements/test_wrong_named_keyword.py]: Let's test more: `!`, `&a`, `()`, `-b`, `+1`, `#FFF`

File: tests/test_visitors/test_ast/test_statements/test_wrong_named_keyword.py [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+
+@pytest.mark.parametrize('code', [
+    '_list = [1, 2]',
+    '_dict = {"end": " "}',
+    'print(*_list, **_dict)',
+    'print(end="1", **{"a": 1})',
+    'filter(**{User.USERNAME_FIELD: username})',  # noqa: P103
+    'filter(**{"a2": 1, _: 2})',
+    'filter(**{"a": 1, b: 2})',
+    'filter(**{"a": 1, call(): 2})',
```
Comment (sobolevn) [inline tests/test_visitors/test_ast/test_statements/test_wrong_named_keyword.py]: Let's test `"_a"` and `"a_"`

File: wemake_python_styleguide/violations/refactoring.py [hunk 4]
```diff
@@ -37,6 +37,7 @@
    ImplicitSumViolation
    FalsyConstantCompareViolation
    WrongIsCompareViolation
+   WrongNamedKeywordViolation
```
Comment (sobolevn) [inline wemake_python_styleguide/violations/refactoring.py]: This is a best-practice. Please, move the violation.

File: tests/test_visitors/test_ast/test_statements/test_wrong_named_keyword.py (hunk 2 above)
Comment (sobolevn) [reply tests/test_visitors/test_ast/test_statements/test_wrong_named_keyword.py]: And also `"some.value"`, `"Username[FIELD]"`, `"call()"`

File: tests/test_checker/test_noqa.py (hunk 1 above)
Comment (ea-volkov) [reply tests/test_checker/test_noqa.py]: Done

File: tests/test_visitors/test_ast/test_statements/test_wrong_named_keyword.py (hunk 2 above)
Comment (ea-volkov) [reply tests/test_visitors/test_ast/test_statements/test_wrong_named_keyword.py]: Done

File: tests/test_visitors/test_ast/test_statements/test_wrong_named_keyword.py (hunk 3 above)
Comment (ea-volkov) [reply tests/test_visitors/test_ast/test_statements/test_wrong_named_keyword.py]: Done

File: wemake_python_styleguide/violations/refactoring.py (hunk 4 above)
Comment (ea-volkov) [reply wemake_python_styleguide/violations/refactoring.py]: Done

**Comment by sobolevn** [comment]
It works! https://github.com/wemake-services/wemake-python-styleguide/commit/94479b6009d1166d37b5c018187de544fc952e4e/checks?check_suite_id=247256538

**Comment by sobolevn** [comment]
Thanks a lot, @ea-volkov! We are now working on `0.13`!
Please, feel free to take part in hacktoberfest: https://hacktoberfest.digitalocean.com

**Pull request merged**